cibuildwheel/macos.py

```diff
@@ -53,19 +55,42 @@ def call(args, env=None, cwd=None, shell=False):
-        if python_package_identifier not in installed_system_packages:
-            # download the pkg
-            call(['curl', '-L', '-o', '/tmp/Python.pkg', config.url])
-            # install
-            call(['sudo', 'installer', '-pkg', '/tmp/Python.pkg', '-target', '/'])
-            # patch open ssl
-            if config.version == '3.5':
-                call(['curl', '-fsSLo', '/tmp/python-patch.tar.gz', 'https://github.com/mayeut/patch-macos-python-openssl/releases/download/v1.0.2t/patch-macos-python-%s-openssl-v1.0.2t.tar.gz' % config.version])
-                call(['sudo', 'tar', '-C', '/Library/Frameworks/Python.framework/Versions/%s/' % config.version, '-xmf', '/tmp/python-patch.tar.gz'])
-
-        installation_bin_path = '/Library/Frameworks/Python.framework/Versions/{}/bin'.format(config.version)
+        if config.identifier.startswith('cp'):
+            # if this version of python isn't installed, get it from python.org and install
+            python_package_identifier = 'org.python.Python.PythonFramework-%s' % config.version
+            if python_package_identifier not in installed_system_packages:
+                # download the pkg
+                call(['curl', '-L', '-o', '/tmp/Python.pkg', config.url])
+                # install
+                call(['sudo', 'installer', '-pkg', '/tmp/Python.pkg', '-target', '/'])
+                # patch open ssl
+                if config.version == '3.5':
+                    call(['curl', '-fsSLo', '/tmp/python-patch.tar.gz', 'https://github.com/mayeut/patch-macos-python-openssl/releases/download/v1.0.2t/patch-macos-python-%s-openssl-v1.0.2t.tar.gz' % config.version])
+                    call(['sudo', 'tar', '-C', '/Library/Frameworks/Python.framework/Versions/%s/' % config.version, '-xmf', '/tmp/python-patch.tar.gz'])
+
+            installation_bin_path = '/Library/Frameworks/Python.framework/Versions/{}/bin'.format(config.version)
+        elif config.identifier.startswith('pp'):
+            pypy_tar_bz2 = config.url.rsplit('/', 1)[-1]
+            assert pypy_tar_bz2.endswith(".tar.bz2")
+            pypy_base_filename = os.path.splitext(os.path.splitext(pypy_tar_bz2)[0])[0]
+            installation_bin_path = os.path.join('/tmp', pypy_base_filename, 'bin')
+            if not os.path.exists(installation_bin_path):
+                call(['curl', '-L', '-o', os.path.join("/tmp", pypy_tar_bz2), config.url])
+                call(['tar', '-C', '/tmp', '-xf', os.path.join("/tmp", pypy_tar_bz2)])
+                pypy_executable = 'pypy' if config.version[0] == '2' else 'pypy3'
+                python_symlink = 'python' if config.version[0] == '2' else 'python3'
+                os.symlink(os.path.join(installation_bin_path, pypy_executable), os.path.join(installation_bin_path, python_symlink))
+                call([os.path.join(installation_bin_path, pypy_executable), '-m', 'ensurepip'])
```

Comment:
Why is a call to `ensurepip` required ?
`pip` gets installed with `call(['python', get_pip_script, '--no-setuptools', '--no-wheel'], env=env)` a few lines after. Isn't that working on its own ?

Reply:
Good point. I saw the `os.symlink` calls for `pip` afterwards, missed the `get-pip.py` after that, and assumed `pip` would be installed already by the `.pkg` installers. Removed the call to `ensurepip`.

Reply:
Huh

> `ImportError: No module named setuptools`

Better add that line again, it seems? :-/

Reply:
(I just force pushed, but here's the log from macOS on Travis CI: https://travis-ci.org/joerick/cibuildwheel/jobs/617454428)

Reply:
Unless you intend to support python2.6, I think `-ensurepip` is the better way to do this in the general case (both pypy and cpython) anyway.

Reply:
@mattip For some reason, it seems we couldn't do that on Windows, though. If I recall correctly, I tried it, but it didn't work with NuGet's installation

Reply:
If I good remember nuget instaltion has pip? Or not always?

Reply:
>> ImportError: No module named setuptools

> Better add that line again, it seems? :-/

On macOS, it seemed redundant to do `ensurepip` then `python get-pip.py`. The latter is needed because TLS 1.2 support was  messy with cpython 3.4 (now dropped) and cpython 3.5 (still used). The error seen in the logs comes from `delocate` installation I guess. Breaking the installation in 2 parts (`setuptools wheel` then `delocate`) should solve the issue.

Reply:
@mayeut Actually, let's just try removing `--no-setuptools` and `--no-wheel` from the `get-pip.py` call. Do you know/remember why that was there?

Reply:
Huh. I used to be able to restart PR builds on Travis CI, I thought? Anyway; failure seems unrelated to the latest forced push, so this works, it seems, without `ensurepip` now.

Reply:
> Do you know/remember why that was there?

I don't know /remember.

Reply:
@joerick Do you still remember? If not, I suggest to keep it like this :-)

Reply:
I don't remember either :P I'm guessing it's not an important detail - we always just want the latest release anyway.